### PR TITLE
enh(api): Remove possibility of updating Vault configuration and storage name

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to remove the possibility of updating Vault configuration and storage name

**Fixes** # MON-16974

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Security/Vault/Domain/Model/NewVaultConfiguration.php [hunk 1]
```diff
@@ -34,9 +34,11 @@ class NewVaultConfiguration
 {
     public const MIN_LENGTH = 1;
     public const MAX_LENGTH = 255;
+    public const NAME_MAX_LENGTH = 50;
     public const MIN_PORT_VALUE = 1;
     public const MAX_PORT_VALUE = 65535;
     public const SALT_LENGTH = 128;
+    public const NAME_VALIDATION_REGEX = '/^[\w+\-_\/]*$/';
```
Comment (abuathier) [inline centreon/src/Core/Security/Vault/Domain/Model/NewVaultConfiguration.php:41]: `\w` iq equivalent o `[A-Za-z0-9_]`, then please remove the superfluous `_`. 

```suggestion
    public const NAME_VALIDATION_REGEX = '/^[\w+\-\/]*$/';
```

File: centreon/src/Core/Security/Vault/Domain/Model/VaultConfiguration.php [hunk 2]
```diff
@@ -33,9 +33,11 @@ class VaultConfiguration
 {
     public const MIN_LENGTH = 1;
     public const MAX_LENGTH = 255;
+    public const NAME_MAX_LENGTH = 50;
     public const MIN_PORT_VALUE = 1;
     public const MAX_PORT_VALUE = 65535;
     public const SALT_LENGTH = 128;
+    public const NAME_VALIDATION_REGEX = '/^[\w+\-_\/]*$/';
```
Comment (abuathier) [inline centreon/src/Core/Security/Vault/Domain/Model/VaultConfiguration.php:40]:
```suggestion
    public const NAME_VALIDATION_REGEX = NewVaultConfiguration::NAME_VALIDATION_REGEX;
```

File: centreon/tests/php/Core/Security/Vault/Domain/Model/NewVaultConfigurationTest.php [hunk 3]
```diff
@@ -34,7 +34,10 @@
 for ($index = 0; $index <= NewVaultConfiguration::MAX_LENGTH; $index++) {
     $invalidMaxLengthString .= 'a';
 }
-
+$invalidNameMaxLengthString = '';
+for ($index = 0; $index <= NewVaultConfiguration::NAME_MAX_LENGTH; $index++) {
+    $invalidNameMaxLengthString .= 'a';
+}
```
Comment (a-launois) [inline centreon/tests/php/Core/Security/Vault/Domain/Model/NewVaultConfigurationTest.php:40]:
```suggestion
$invalidNameMaxLengthString = str_repeat('a', NewVaultConfiguration::NAME_MAX_LENGTH+ 1);
```

Comment (abuathier) [inline centreon/tests/php/Core/Security/Vault/Domain/Model/NewVaultConfigurationTest.php:40]: The use of [str_repeat](https://www.php.net/manual/fr/function.str-repeat.php) is probably easier to understand ?
```suggestion
$invalidNameMaxLengthString = str_repeat('a', 1 + NewVaultConfiguration::NAME_MAX_LENGTH);
```

**Review by abuathier — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Pull request merged**